nautobot/extras/models/relationships.py

```diff
@@ -257,10 +257,8 @@ def required_related_objects_errors(
                     and initial_data.get(relation, {}).get(opposite_side, {}) == {}
                     and not relationships_key_specified
                 ):
-                    if (
-                        RelationshipAssociation.objects.filter(**{f"{relation.required_on}_id": instance.pk}).count()
-                        > 0
-                    ):
+                    filter_kwargs = {"relationship": relation, f"{relation.required_on}_id": instance.pk}
+                    if RelationshipAssociation.objects.filter(**filter_kwargs).count() > 0:
```

Comment:
This should be an `exists()` query instead:

```suggestion
                    if RelationshipAssociation.objects.filter(**filter_kwargs).exists():
```